tango_simlib/tango_yaml_tools/base.py

```diff
@@ -39,57 +39,51 @@ def _build_yaml(self):
             }
         ]
 
-        for command in self.parser.get_device_command_metadata().values():
-            command_data = {
-                "name": command["name"],
-                "dtype_in": command["dtype_in"].name,
-                "dtype_out": command["dtype_out"].name,
-            }
-            for key in ["doc_out", "doc_in", "inherited", "disp_level"]:
-                if key in command:
+        command_values = self.parser.get_device_command_metadata().values()
+        command_values = sorted(command_values, key=lambda x: x["name"])
+        for command in command_values:
+            command_keys = sorted(command.keys())
+            command_keys.insert(0, command_keys.pop(command_keys.index("name")))
+            command_data = {}
+            for key in command_keys:
+                if key in ["dtype_in", "dtype_out"]:
+                    command_data[key] = command[key].name
+                else:
                     command_data[key] = command[key]
             data_dict[0]["meta"]["commands"].append(command_data)
 
-        for attr in self.parser.get_device_attribute_metadata().values():
-            attr_data = {"name": attr["name"]}
-            if "data_type" in attr:
-                attr_data["data_type"] = attr["data_type"].name
-            if "data_format" in attr:
-                attr_data["data_format"] = attr["data_format"].name
-            if "disp_level" in attr:
-                attr_data["disp_level"] = attr["disp_level"].name
-            # pylint insists on this spacing
-            for key in [
-                    "delta_val",
-                    "period",
-                    "display_unit",
-                    "standard_unit",
-                    "unit",
-                    "max_dim_y",
-                    "max_dim_x",
-                    "label",
-                    "max_value",
-                    "min_alarm",
-                    "max_warning",
-                    "description",
-                    "format",
-                    "delta_t",
-                    "max_alarm",
-                    "min_value",
-                    "inherited",
-                    "min_warning",
-                    "writable",
-                    "writable_attr_name",
-            ]:
-                if key in attr:
+        attr_values = self.parser.get_device_attribute_metadata().values()
+        attr_values = sorted(attr_values, key=lambda x: x["name"])
+        for attr in attr_values:
+            attr_keys = sorted(attr.keys())
+            attr_keys.insert(0, attr_keys.pop(attr_keys.index("name")))
+            attr_data = {}
+            for key in attr_keys:
+                if key in ["data_format", "data_type", "disp_level"]:
+                    attr_data[key] = attr[key].name
+                elif key in [
+                        "name",
+                        "display_unit",
+                        "standard_unit",
```

Comment:
I see `unit` has been removed, while we kept the other two.  I don't know which of the 3 are more commonly used, hopefully we aren't losing any useful info.

Reply:
I missed a few items, back to the original list